src/Mod/PartDesign/Gui/Command.cpp

```diff
@@ -1122,6 +1122,20 @@ CmdPartDesignAdditiveLoft::CmdPartDesignAdditiveLoft()
 
 void CmdPartDesignAdditiveLoft::activated(int iMsg)
 {
+	App::Document *doc = getDocument();
+	PartDesign::Body *pcActiveBody = PartDesignGui::getBody(
+		/*messageIfNot = */ PartDesignGui::assureModernWorkflow(doc));
```

Comment:
The loft (and sweep) are not available in legacy workflow, hence they should only work with bodies, never without and hence never in legacyworkflow. hence it is better to remove the workflow query and always set messageIfnot to true.